### PR TITLE
Add missing System.Collections API docs

### PR DESCRIPTION
**Pull request by eiriktsarpalis**
Fix https://github.com/dotnet/runtime/issues/72407.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -240,3 +240,6 @@ samples/framework/docker/MVCRandomAnswerGenerator/containerImage
 .DS_Store
 _dependentPackages/
 !/xml/System.Net.Cache/
+
+# Ignore ionide caches
+.ionide/
```
Comment (eiriktsarpalis) [inline .gitignore:245]: This has been added intentionally, tells git to ignore the cache folder created by the VS Code F# extension.

**Comment by carlossanlop** [comment]
@eiriktsarpalis Thanks for submitting! Here is the latest list as of Preview7. I see 3 APIs that were added in that preview, which also need documentation (the first 3). Can you include them in this PR, please?

| API                                                                                                                                                                                                                                | Summary  | Parameters | TypeParameters | ReturnValue |
|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------|------------|----------------|-------------|
| [M:System.Collections.Generic.SortedList2.GetKeyAtIndex(System.Int32)](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Collections.Generic/SortedList2.xml)                                                         | Missing  | Missing    | NA             | Missing     |
| [M:System.Collections.Generic.SortedList2.GetValueAtIndex(System.Int32)](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Collections.Generic/SortedList2.xml)                                                       | Missing  | Missing    | NA             | Missing     |
| [M:System.Collections.Generic.SortedList2.SetValueAtIndex(System.Int32,1)](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Collections.Generic/SortedList2.xml)                                                     | Missing  | Missing    | NA             | NA          |
| [M:System.Collections.Immutable.ImmutableArray1.AddRange1(System.Collections.Immutable.ImmutableArray{0})](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Collections.Immutable/ImmutableArray1.xml)               | Present  | Present    | Missing        | Present     |
| [M:System.Collections.Immutable.ImmutableArray1.AddRange1(0[])](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Collections.Immutable/ImmutableArray1.xml)                                                          | Present  | Present    | Missing        | Present     |
| [M:System.Collections.Immutable.ImmutableArray1.Builder.AddRange1(System.ReadOnlySpan{0})](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Collections.Immutable/ImmutableArray1+Builder.xml)                       | Present  | Present    | Missing        | NA          |
| [M:System.Collections.Immutable.ImmutableList1.Builder.Remove(0,System.Collections.Generic.IEqualityComparer{0})](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Collections.Immutable/ImmutableList1+Builder.xml) | Present  | Present    | NA             | Missing     |

**Comment by eiriktsarpalis** [comment]
@carlossanlop This PR already adds documentation for all the items in that list, I don't believe anything has been left out.

**Review by carlossanlop — COMMENTED** [review]
LGTM so far. I only wonder if the `TDerived` description could be reworder in a simpler way, but I don't think it's blocking.

Only thing left to address is the request to include the other 3 APIs.

**Review by carlossanlop — APPROVED** [review]
LGTM, thanks! Let's just wait for the CI.

**Pull request merged**